draft-ietf-sidrops-aspa-verification.xml

```diff
@@ -448,227 +348,153 @@ hop(AS(i), AS(j)) =  / Else, "Provider+" if the VAP-SPAS entry
-            Call it u_min.
-            If no such u_min exists, set u_min = N+1.
-            Find the highest value of v (N-1 &ge; v &ge; 1) for which hop(AS(v+1), AS(v)) = "Not Provider+".
-            Call it v_max.
-            If no such v_max exists, then set v_max = 0.
-            If u_min &le; v_max, then the procedure halts with the outcome "Invalid".
-            Else, continue.
+            If min_up_rump is less than N the procedure halts with the outcome 'Unknown'.
+            Else, the procedure halts with the outcome "Valid".
+          </t>
+        </list>
+      </t>
+    </section>
+
+    <section title="Algorithm for Downstream Paths" anchor="Downflow">
+      <t>
+        #TODO: move mutual transit in the above section. It is incorrect to speak about it as provider. It and should be processed at the prefix based level upon its policy.
+
+        The downstream verification algorithm described here is applied when a route is received from a transit provider.
+      </t>
+      <t>
+        <list style="numbers">
+          <t>
+            If the AS_PATH has an AS_SET, then the procedure halts with the outcome "Invalid".
           </t>
           <t>
-            Up-ramp: For 2 &le; i &le; N, determine the largest K such that hop(AS(i-1), AS(i)) = "Provider+" for each i in the range 2 &le; i &le; K.
-            If such a largest K does not exist, then set K = 1.
+            Collapse prepends in the AS_SEQUENCE(s) in the AS_PATH (i.e., keep only the unique AS numbers).
+            Let the resulting ordered sequence be represented by {AS(N), AS(N-1), ..., AS(2), AS(1)}, where AS(1) is the first-added (i.e., origin) AS and AS(N) is the last-added AS and neighbor to the receiving/validating AS.
           </t>
-<!--
-            <t>
-	If K = N, then the procedure halts with the outcome "Valid".
-	Else, continue.
-            </t>
--->
           <t>
-            Down-ramp: For N-1 &ge; j &ge; 1, determine the smallest L such that hop(AS(j+1), AS(j)) = "Provider+" for each j in the range N-1 &ge; j &ge; L.
-            If such smallest L does not exist, then set L = N.
+            If 1 &le; N &le; 2, then the procedure halts with the outcome "Valid".
+            Else, continue.
+          </t>
+          <t>
+            At this step, N &ge; 3.
+            If the sum of max_up_rump and max_down_rump is less than N the procedure halts with the outcome 'Invalid'.
           </t>
           <t>
-            If L-K &le; 1, then the procedure halts with the outcome "Valid".
-            Else, the procedure halts with the outcome "Unknown".
+            If the sum of min_up_rump and min_down_rump is less than N the procedure halts with the outcome 'Unknown'.
           </t>
         </list>
       </t>
+    </section>
+      
+    <section anchor="mitigation" title="Configuration of a Mitigation Policy">
```

Comment:
Maybe worth moving it into deployment recommendations section.

And I think there we already have a section on what to do with **invalid**, **valid**, **unknown**

Reply:
I moved it from deployment considerations intentionally, because mitigation is the key part of the whole idea.

Reply:
But I would figure that the admin who deploys this in the future will open the RFC and Look into Deployment Recommendations, and then wounder why he doesn't find it there.

Reply:
@mitradir, i agree that the mitigation is the key part of this document, but it is not well placed in this sectionn

the section title says "AS_PATH Verification". the subsection title is "Configuration of a Mitigation Policy" -- this is not verification ;). now, we could change the section title to "AS_PATH Verification and Mitigation". then we have the problem that it might appear that this document requires filtering of invalids (because implementing the algorithm is required), even though the intro of "Mitigation Policy" explicitly notes "verification using ASPA is at the discretion of network operator".

a compromise: we could add another \section{Mitigation Policy}.

Reply:
Sounds good to me, I will move it to a separate section.